core/types/transaction_test.go

```diff
@@ -217,5 +220,19 @@ func TestTransactionJSON(t *testing.T) {
 		if tx.ChainId().Cmp(parsedTx.ChainId()) != 0 {
 			t.Errorf("invalid chain id, want %d, got %d", tx.ChainId(), parsedTx.ChainId())
 		}
+		if tx.L1MessageSender() != nil && parsedTx.L1MessageSender() != nil && *tx.L1MessageSender() != *parsedTx.L1MessageSender() {
```

Comment:
Noting that it seems like we would not catch an error here if `tx.L1MessageSender() != nil` but `parsedTx.L1MessageSender() == nil`.  Not positive it's an issue but thought worth noting.

Reply:
If `parsedTx.L1MessageSender() == nil`, evaluating `*parsedTx.L1MessageSender()` will throw a Null Pointer Exception, since we're trying to access the memory address pointed to by `nil`.

Reply:
Ahh great, and presumably all three args are evaluated even if the second one is false.   Got it!

Reply:
No, it should eagerly exit the evaluation. Sorry, I misread your comment. Yes, I'll add a check if one is nil but the other is not.